packages/api/src/controllers/access-control.test.ts

```diff
@@ -0,0 +1,175 @@
+import serverPromise, { TestServer } from "../test-server";
+import { TestClient, clearDatabase, setupUsers } from "../test-helpers";
+import { SigningKey, SigningKeyResponsePayload, User } from "../schema/types";
+import { WithID } from "../store/types";
+import { Asset } from "../schema/types";
+import { v4 as uuid } from "uuid";
+import { db } from "../store";
+import { generateUniquePlaybackId } from "./generate-keys";
+
+// includes auth file tests
+
+let server: TestServer;
+let mockAdminUserInput: User;
+let mockNonAdminUserInput: User;
+
+// jest.setTimeout(70000)
+
+beforeAll(async () => {
+  server = await serverPromise;
+
+  mockAdminUserInput = {
+    email: "user_admin@gmail.com",
+    password: "x".repeat(64),
+  };
+
+  mockNonAdminUserInput = {
+    email: "user_non_admin@gmail.com",
+    password: "y".repeat(64),
+  };
+});
+
+afterEach(async () => {
+  await clearDatabase(server);
+});
+
+describe("controllers/signing-key", () => {
+  describe("basic CRUD with JWT authorization", () => {
+    let client: TestClient;
+    let adminUser: User;
+    let adminToken: string;
+    let nonAdminUser: User;
+    let nonAdminToken: string;
+    let otherUserToken: string;
+    let signingKey: WithID<SigningKey>;
+    let gatedPlaybackId: string;
+    let publicPlaybackId: string;
+    let gatedAsset: WithID<Asset>;
+
+    beforeEach(async () => {
+      ({ client, adminUser, adminToken, nonAdminUser, nonAdminToken } =
+        await setupUsers(server, mockAdminUserInput, mockNonAdminUserInput));
+      client.jwtAuth = nonAdminToken;
+      let created: SigningKeyResponsePayload = await client
+        .post("/access-control/signing-key")
+        .then((res) => res.json());
+      let res = await client.get(`/access-control/signing-key/${created.id}`);
+      signingKey = await res.json();
+      const res2 = await client.post("/stream", {
+        name: "test",
+        playbackPolicy: {
+          type: "signed",
```

Comment:
Ah, one more thing... Can we change this playback policy type to `jwt`? It will be confusing to add other "signature" methods later (like web3 signature) but have only the JWT playback policy being called "signed".

Can be made on a separate PR as it's unrelated to this new API, but IMO it'd be really desirable.

Reply:
Cool, changing it to `jwt`